packages/dds/tree/src/feature-libraries/sequence-change-family/compose.ts

```diff
@@ -0,0 +1,246 @@
+        if (baseMark === undefined) {
+            // We have reached an region of the field that the base change does not affect.
+            // We therefore adopt the new mark as is.
+            factory.push(clone(newMark));
+        } else if (isAttachGroup(newMark)) {
+            // Content that is being attached by the new changeset cannot interact with base changes.
+            // Note that attach marks from different changesets can only target the same gap if they are concurrent.
+            // If two concurrent changesets A and B were targeting the same gap, the changeset that is sequenced later
+            // would bet rebased over the first, resulting in a Skip mark being introduced in the later changeset.
+            // Since compose only deals with changes that are rebased in this way,
+            // we don't have to deal with tie-breaking issues.
+            factory.pushContent(clone(newMark));
+            baseIter.push(baseMark);
+        } else if (isReattach(newMark)) {
+            // Content that is being re-attached by the new changeset can interact with base changes.
+            // This can happen in two cases:
+            // - The base change contains the detach that the re-attach is the inverse of.
+            // - The base change contains a tombstone for the detach that the re-attach is the inverse of.
+            // We're ignoring these cases for now. The impact of ignoring them is that the relative order of
+            // reattached content and concurrently attached content is not preserved.
+            // TODO: properly compose reattach marks with their matching base marks if any.
+            factory.pushContent(clone(newMark));
+            baseIter.push(baseMark);
+        } else if (isDetachMark(baseMark)) {
+            // Content that is being detached by the base changeset can interact with the new changes.
+            // This can happen in two cases:
+            // - The new change contains reattach marks for this detach. (see above)
+            // - The new change contains tombs for this detach.
+            // We're ignoring these cases for now. The impact of ignoring them is that the relative order of
+            // reattached content and concurrently attached content is not preserved.
+            // TODO: properly compose detach marks with their matching new marks if any.
+            factory.pushContent(baseMark);
+            newIter.push(newMark);
+        } else if (isTomb(baseMark) || isGapEffectMark(baseMark) || isTomb(newMark) || isGapEffectMark(newMark)) {
+            // We don't currently support Tomb and Gap marks (and don't offer ways to generate them).
+            fail("TODO: support Tomb and Gap marks");
+        } else {
+            // If we've reached this branch then `baseMark` and `newMark` start at the same location
+            // in the document field at the revision after the base changes and before the new changes.
```

Comment:
This comment makes me wish we had a good term for the revision which is sandwiched by the changesets we're composing. Unfortunately I don't have a good suggestion.

Reply:
Yeah, I had the same though. Some terms I considered but was not thrilled with:
* Interstitial revision
* Joining revision
* Common revision